### PR TITLE
Support LowLevelCallable

### LINKED ISSUE #2995: Support passing in LowLevelCallable as argument to jitted function
With the recent support for passing in numba jitted functions as arguments (https://github.com/numba/numba/pull/2801), it would also be nice to also support passing in [`scipy.LowLevelCallable`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.LowLevelCallable.html) objects. These have become common in the scipy ecosystem, and would be useful for downstream projects considering numba adoption.

### PR DESCRIPTION
**Pull request by sklam**
Fix #2995

- [x] add ctypes lowlevelcallable
- [x]  add cffi lowlevelcallable
- [ ]  support .user_data

### CODE REVIEW

File: numba/types/scipytypes.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+"""
+Typing for scipy specific features
+"""
+from numba import types
+
```
Comment (seibert) [inline numba/types/scipytypes.py]: This creates an import cycle, which appears to be handled properly in Python 3, but not Python 2.

**Comment by dhirschfeld** [comment]
I'm very keen on this functionality and it seems like it's very nearly there so I'm curious if it will make the cutoff for 0.40.0?

**Comment by dhirschfeld** [comment]
Ping! It seems like this PR is most of the way there. Is there anything in particular holding it up?

I'm hopeful that it will resolve #2578 as well.

**Comment by jfbourdon** [comment]
It would be sad to see this being closed... but well, as a simple occasional user I'm really not the one who can help and I won't pretend understanding a single line of code behind this PR!

**Pull request closed without merging**